### PR TITLE
Moveit simple ikfast collision check wrapper

### PR DESCRIPTION
**Pull request by MauleshSTrivedi**
First commit. I will be adding the updated URDFs with the unit tests for custom_tool collision check.

### CODE REVIEW

File: moveit_simple/cfg/moveit_simple_dynamic_reconfigure_.params [hunk 1]
```diff
@@ -4,6 +4,6 @@ PACKAGE = "moveit_simple"
 from rosparam_handler.parameter_generator_catkin import *
 gen = ParameterGenerator()
 
-gen.add("speed_modifier", paramtype="double", description="Speed Modifier for Execution", min=0.01, max=1.0, default=1.0, configurable=True)
+gen.add("speed_modifier", paramtype="double", description="Speed Modifier for Execution", min=0.01, max=1.0, default=0.25, configurable=True)
```
Comment (shaun-edwards) [inline moveit_simple/cfg/moveit_simple_dynamic_reconfigure_.params]: Set back to 1.0

File: moveit_simple/src/moveit_simple.cpp [hunk 2]
```diff
@@ -63,8 +64,54 @@ Robot::Robot(const ros::NodeHandle & nh, const std::string &robot_description,
...
+                      const std::string &kinematics_base_link,
+                      const std::string &kinematics_tip_link):
+  tf_buffer_(),
+  tf_listener_(tf_buffer_),
+  nh_(nh),
+  params_(ros::NodeHandle("~/moveit_simple")),
+  speed_modifier_(1.0),
+  dynamic_reconfig_server_(ros::NodeHandle("~/moveit_simple"))
+{
+  ROS_INFO_STREAM("Loading MoveIt objects based on, robot description: " << robot_description
```
Comment (shaun-edwards) [inline moveit_simple/src/moveit_simple.cpp]: Create an init function.

File: moveit_simple/src/moveit_simple.cpp [hunk 3]
```diff
@@ -63,8 +64,54 @@ Robot::Robot(const ros::NodeHandle & nh, const std::string &robot_description,
...
+                        nh_.getNamespace() + "/rviz_visual_tools",
+                        robot_model_ptr_));
+  virtual_visual_tools_->loadRobotStatePub(nh_.getNamespace()
+                                    + "/display_robot_state");
+
+  // Dynamic Reconfigure Parameters with rosparam_handler
+  params_.fromParamServer();
+  dynamic_reconfig_server_.setCallback(boost::bind(&Robot::reconfigureRequest, this, _1, _2));
+
+  kinematics_base_link_override = kinematics_base_link;
```
Comment (shaun-edwards) [inline moveit_simple/src/moveit_simple.cpp]: Instead cache these values as either overrides or the `joint_group_->getSolverInstance()->getTipFrame();`, etc...

**Review by shaun-edwards — CHANGES_REQUESTED** [review]

**Comment by MauleshSTrivedi** [comment]
@shaun-edwards Modified according to request. Please review. Thanks. :)

**Comment by MauleshSTrivedi** [comment]
Moving this to #47

**Pull request closed without merging**